alts/src/main/java/io/grpc/alts/AltsChannelBuilder.java

```diff
@@ -115,13 +120,25 @@ protected NettyChannelBuilder delegate() {
 
   @Override
   public ManagedChannel build() {
-    CheckGcpEnvironment.check(enableUntrustedAlts);
+    Status status = Status.OK;
+    if (!CheckGcpEnvironment.isOnGcp()) {
+      if (enableUntrustedAlts) {
+        logger.log(
+            Level.WARNING,
+            "Untrusted ALTS mode is enabled and we cannot guarantee the trustworthiness of the "
+                + "ALTS handshaker service");
+      } else {
+        status =
+            Status.FAILED_PRECONDITION.withDescription(
+                "ALTS is only allowed to run on Google Cloud Platform");
+      }
+    }
+
     TcpfFactory tcpfFactory = new TcpfFactory();
     InternalNettyChannelBuilder.setDynamicTransportParamsFactory(delegate(), tcpfFactory);
-
     tcpfFactoryForTest = tcpfFactory;
 
-    return new AltsChannel(delegate().build());
+    return delegate().intercept(new AltsClientInterceptor(status)).build();
```

Comment:
Only add the interceptor when it will be helpful (like in the else case above).

You should add a note that you can't call `build()` multiple times.